pkg/entities/record.go

```diff
@@ -76,11 +79,12 @@ type templateRecord struct {
 func NewTemplateRecord(count uint16, id uint16) *templateRecord {
 	return &templateRecord{
 		&baseRecord{
-			buff:       bytes.Buffer{},
-			len:        0,
-			fieldCount: count,
-			templateID: id,
-			elements:   make([]*InfoElementWithValue, 0),
+			buff:        bytes.Buffer{},
+			len:         0,
+			fieldCount:  count,
+			templateID:  id,
+			elementList: make([]*InfoElementWithValue, 0),
```

Comment:
could you change the name of elementList like orderedElementList or something that signifies why you need a list along with the map?

Reply:
Sure. Updated. Thanks!